### PR TITLE
Fix types module name

### LINKED ISSUE #354: Typescript typigns not working
## Expected Behavior
You should be able to import the module with typescript: `import Vonage from "@vonage/server-sdk";`

## Current Behavior
There are multiple issues not allowing it to import correctly:
1. The module name in `typings/index.d.ts` is wrong: It is currently `vonage`, but needs to be `@vonage/server-sdk` to read `declare module '@vonage/server-sdk' { ... }`
2. `VonageApiError` is not defined, which appears in `export interface VerifyError extends VonageApiError`. Renaming `NexmoApiError` to `VonageApiError` fixes this
3. When finally importing and trying to create a new instance, it does not work:
```
import Vonage from "@vonage/server-sdk";
const vonage = new Vonage({ ... });
```
gives
```
TypeError: server_sdk_1.default is not a constructor
    at new Message (/home/sorunome/repos/famedly/contact-finder/build/src/message.js:46:23)
    at /home/sorunome/repos/famedly/contact-finder/build/src/index.js:123:25
    at Generator.next (<anonymous>)
    at fulfilled (/home/sorunome/repos/famedly/contact-finder/build/src/index.js:21:58)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

A possible solution for this is to
```
import * as Vonage from "@vonage/server-sdk";
const vonage = new Vonage({ ... });
```
But then you aren't able to use the `Vonage` type, which makes typings kinda useless
(still thinking of a possible solution for this one, will edit in if found)

## Possible Solution
See above

## Steps to Reproduce (for bugs)
Try to use library with typescript

## Your Environment

* SDK Version: 2.10.1
* Node Version: (e.g. Node 10.18): 14.9.0
* Operating System and version: Archlinux

### PR DESCRIPTION
**Pull request by kellyjandrews**
## Description
Updated `vonage` to `@vonage/server-sdk` and `NexmoApiError` to `VonageApiError`

## Motivation and Context
Fixes #354 

## Testing Details
No changes

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -19,9 +19,9 @@ declare module 'vonage' {
         privateKey?: string;
     }
 
-    export interface NexmoApiError {
-        body: { [key: string]: any };
-        headers?: { [key: string]: any };
+    export interface VonageApiError {
```
Comment (dragonmantank) [inline typings/index.d.ts:22]: Do we need to change any internal references from `NexmoApiError` to `VonageApiError` as well?

**Review by dragonmantank — CHANGES_REQUESTED** [review]
Module name change looks fine, just a question on the interface name change

File: typings/index.d.ts (hunk 1 above)
Comment (kellyjandrews) [reply typings/index.d.ts:22]: I believe all other references were changed.

**Review by dragonmantank — APPROVED** [review]
LGTM :+1:

**Pull request merged**